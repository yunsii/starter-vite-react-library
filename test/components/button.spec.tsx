import React from 'react'
import renderer from 'react-test-renderer'
import { expect, it } from 'vitest'

import { Button } from '../../src'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

it('link changes the class when hovered', () => {
  const component = renderer.create(<Button>Anthony Fu</Button>)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
