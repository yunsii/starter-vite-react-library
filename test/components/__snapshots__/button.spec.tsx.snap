// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`link changes the class when hovered 1`] = `
<button
  type="button"
>
  Yuns
</button>
`;
